gradle/release.gradle

```diff
@@ -106,121 +105,147 @@ task("releaseNeeded") {
-            .rollback { run "git", "reset", "--hard", "HEAD~1" }
+task resetReleaseNotes(type: Exec) {
+    description = "Removes release notes commit by resetting head by one commit (--hard)."
+    commandLine "git", "reset", "--hard", "HEAD~1"
+}
 
-    step("create new version tag") { createTag(buildInfo, "v${currentVersion}".toString()) }
-            .rollback { run "git", "tag", "-d", "v${currentVersion}".toString()}
+task tagNewVersion(type: Exec) {
+    commandLine "git", "tag", "-a", "v${currentVersion}", "-m", "Create tag v${currentVersion} ${buildInfo}"
+}
 
-    step("commit incremented version on $System.env.TRAVIS_BRANCH") { commitIncrementedVersion(currentVersion, buildInfo, project.versionFile) }
-            .rollback { run "git", "reset", "--hard", "HEAD~1" }
+task removeNewTag(type: Exec) {
+    commandLine "git", "tag", "-d", "v${currentVersion}"
+}
 
-    step("push changes to all involved branches") {
-        def pushCommandLine = ["git", "push", pushTarget, System.env.TRAVIS_BRANCH, "v$currentVersion", "-q"]
-        if (dryRun) {
-            pushCommandLine << '--dry-run'
-        }
-        run pushCommandLine
+task bumpVersion {
+    description = "Increments version in 'version.properties' and commits to '$System.env.TRAVIS_BRANCH'"
+    doLast {
+        String nextVersion = project.versionFile.incrementVersion()
+        String message = "Increment version '$currentVersion' -> '$nextVersion' $buildInfo"
+        run "git", "commit", "-m", "$message" as String, "version.properties"
     }
 }
 
-/*
-TODO:
- - declare release steps and then Gradle plugin adds tasks:
-  release, rollbackRelease, releaseStep1, rollbackStep2, releaseSteps1..7
- - improve exception reporting
- - start rolling back from last successful step (e.g. don't rollback the step that failed)
- */
-
-private void commitReleaseNotes(String buildInfo) {
-    def notesFile = project.file("doc/release-notes/official.md")
-    run "git", "add", "$notesFile" as String
-    run "git", "commit", "-m", "Update release notes $buildInfo" as String, "$notesFile" as String
+task unbumpVersion(type: Exec) {
+    description = "Unbumps version in 'version.properties' by resetting head by one commit (--hard)."
+    commandLine "git", "reset", "--hard", "HEAD~1"
 }
 
-private void createTag(String buildInfo, String tag) {
-    String tagMessage = "Create tag ${tag} ${buildInfo}"
-    run "git", "tag", "-a", tag, "-m", "$tagMessage" as String
+task pushChanges(type: Exec) {
+    description = "Pushes changes to '${System.env.TRAVIS_BRANCH}' branch."
+    //!!!Below command _MUST_ be quiet otherwise it exposes GitHub write token!!!
+    def mustBeQuiet = "-q"
+    def pushCommandLine = ["git", "push", "https://szczepiq:${System.env.GH_TOKEN}@github.com/mockito/mockito.git", System.env.TRAVIS_BRANCH, "v$currentVersion", mustBeQuiet]
+    if (releaseTest) {
+        pushCommandLine << '--dry-run'
+    }
+    commandLine pushCommandLine
```

Comment:
Yeah, interesting! So far push has not failed. If it does I'm not sure retrying would help.